src/metabase/api/search.clj

```diff
@@ -369,6 +360,20 @@
                                      query))} :alias_is_required_by_sql_but_not_needed_here]]
       :order-by order-clause})))
 
+(defn- hydrate-user-metadata
+  "Hydrate common-name for last_edited_by and created_by from result."
+  [results]
+  (let [user-ids             (set (flatten (for [result results]
+                                             (remove nil? ((juxt :last_editor_id :creator_id) result)))))
+        user-id->common-name (if (pos? (count user-ids))
+                               (t2/select-pk->fn :common_name [:model/User :id :first_name :last_name :email] :id [:in user-ids])
+                               {})]
+    (mapv (fn [{:keys [creator_id last_editor_id] :as result}]
+            (assoc result
+                   :creator_common_name (get user-id->common-name creator_id)
+                   :last_editor_common_name (get user-id->common-name last_editor_id)))
+          results)))
+
```

Comment:
Will this extra DB query materially slow down search results? I wonder if we should add this as a join in the search results queries themselves, and calculate the common_name with a function reused from the User model namespace.

Reply:
In theory whether it slows down the result by much is hard to say because it'll require us to add retrieve 6 more columns to be able to do this, also for the last_editor_common_name, we will need to join twice -- one against revision, then revision against user.

nevertheless, theory is meaningless without actual benchmarking:
I had this branch: https://github.com/metabase/metabase/compare/ngoc-search-more-metadata...ngoc-search-more-metadata-join-approach?expand=1

where I tried the joining approach. 

I had a setup of 5k cards, 1k models, 1k dashboards, and 100 collections.

I did some benchmarking and both versions have about the same response time on my machine (~190ms) with the current version slightly faster.

Reply:
also, I think the code for current approach is cleaner